File: Atcoder_#320/javascript/b_longest_palindrome.js
function main(input) {
  const isPalindrome = (str) => str === str.split('').reverse().join('');
  
  let maxLength = 0;
  const S = input.trim();

  for (let i = 0; i < S.length; i++) {
    for (let j = i + 1; j <= S.length; j++) {
      const substring = S.slice(i, j);
      if (isPalindrome(substring)) {
        maxLength = Math.max(maxLength, substring.length);
      }
    }
  }

  console.log(maxLength);
}

main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: Atcoder_#320/ruby/b_longest_palindrome.rb
def is_palindrome?(str)
  str == str.reverse
end

def longest_palindrome(s)
  max_length = 0

  (0...s.length).each do |i|
    (i+1..s.length).each do |j|
      substring = s[i...j]
      if is_palindrome?(substring)
        max_length = [max_length, substring.length].max
      end
    end
  end

  max_length
end

s = gets.chomp
puts longest_palindrome(s)
